### PR TITLE
Systematizes text styles [Fixes #2342]

### LINKED ISSUE #2342: Systematize text styles
**Is your feature request related to a problem? Please describe.**

Our text (particularly header) styles are getting a bit messy (shocker - CSS getting unruly!?).

Currently we have a variety of ways we've been setting header styles.

1) in SharedStyledComponents.js:

```
export const H1 = styled.h1`
  ${Mixins.textLevel1}
`
export const H2 = styled.h2`
  ${Mixins.textLevel2}
  @media (max-width: ${(props) => props.theme.breakpoints.m}) {
    font-size: 1.5rem;
  }
`
export const H3 = styled.h3`
  ${Mixins.textLevel3}
`
export const H4 = styled.h4`
  ${Mixins.textLevel4}
`
export const H5 = styled.h5`
  ${Mixins.textLevel5}
`
```

... which imports from

2) in theme.js where we create mixins:

```
export const Mixins = {
  textLevel1: `
    font-size: 3rem;
    line-height: 1.4;
    margin: 2rem 0;
    font-weight: 400;
  `,
  textLevel2: `
    font-size: 2rem;
    line-height: 1.4;
    margin: 2rem 0;
    font-weight: 600;
  `,
  textLevel3: `
    font-size: 1.5rem;
    line-height: 1.4;
    margin: 1.5rem 0 2rem;
    font-weight: 600;
  `,
  textLevel4: `
    font-size: 1.25rem;
    line-height: 1.4;
    font-weight: 400;
    margin-top: 2rem;
  `,
  textLevel5: `
    font-size: 1rem;
    line-height: 1.6;
    font-weight: 400;
    margin-top: 2rem;
  `,
  textLevel6: `
    font-size: .875rem;
    line-height: 1.6;
    font-weight: 400;
    letter-spacing: 0.04em;
    margin: 1.14em 0;
    text-transform uppercase
  `,
  textLevel7: `
    font-size: 1rem;
    line-height: 1.6;
    font-weight: 400;
    margin: 2rem 0 1rem;
  `,
  textLevel8: `
    font-size: .875rem;
    line-height:1.6;
    margin: 1.14em 0;
    font-weight: 400;
  `,
}
```

3. in individual page components, e.g. src/pages/index.js:

```
const H1 = styled.h1`
  font-size: 40px;
  font-weight: 700;
  margin-bottom: 0rem;
  text-align: center;
  @media (max-width: ${(props) => props.theme.breakpoints.s}) {
    font-size: 32px;
  }
`
```

**Describe the solution you'd like**

I suspect this can be simplified.

For one, we don't use the `Mixins` from theme.js much. Should we ditch those?

Instead, perhaps we:
- roll out defaults for all text elements (h1-h6, p, body, etc.) in `GlobalStyle` of theme.js.
- delete the H1-H6 in SharedStyledComponents.js (since those are now the defaults)
- extend the defaults in individual page components when necessary (as we do currently)

**Describe alternatives you've considered**

I welcome your thoughts & ideas @ryancreatescopy @wackerow 

**Additional context**

Surfaced in homepage work: #2124

### PR DESCRIPTION
**Pull request by wackerow**
## Description
Attempt at simplifying header tag styling across site. 
- This eliminates mixins and just adds h1-h5 styling directly to GlobalStyle in theme.js.
- This also eliminates the need for H1-H5 in the SharedStylesComponent, and instances have been removed.
- Some custom local styling had to be applied to a variety of pages, primarily with issues around the h1 and h2 headers.

@ryancreatescopy @samajammin Could use a hand auditing this. Ryan, curious to hear your thoughts on the subtle font-weight changes that you may notice here and there through this... lemme know what you think

## Related Issue
#2342

### CODE REVIEW

File: src/components/Roadmap.js [hunk 1]
```diff
@@ -208,9 +208,9 @@ const Roadmap = () => {
         .
       </p>
       <IssueSection issues={issues.implemented} />
-      <H2>
+      <h2>
```
Comment (Kristiyan96) [inline src/components/Roadmap.js:211]: Use the H2 you defined in src/components/CalloutBanner.js ?

File: src/pages-conditional/wallets/index.js [hunk 2]
```diff
@@ -155,6 +155,10 @@ const CalloutCardContainer = styled(CardContainer)`
   margin-top: 4rem;
 `
 
+const H2 = styled.h2`
```
Comment (Kristiyan96) [inline src/pages-conditional/wallets/index.js:158]: Why defining different H2 styles?

**Review by Kristiyan96 — CHANGES_REQUESTED** [review]
Inconsistent header styles, move them to a single file that defines all of them and import where needed.

**Review by samajammin — APPROVED** [review]
Code LGTM! This doesn't get us all the way there in terms of unifying styles but it's a great 1st step.

@ryancreatescopy mind clicking around this preview PR & spot-checking that these changes didn't disrupt anything major?

**Comment by ryancreatescopy** [comment]
Awesome, so have gone through this and made some tweaks. 

Some thoughts on how I think we should approach this moving forward.

1. The heading styles in the themes should define headings for use in Markdown as these can't be easily edited in a `.md` file. Thus the styling should aim to improve readability in markdown files.
2. We should create a Heading component for use in `.js` pages with less baked-in styling. My suggestion would be to define default sizing, weight, line-height and margin bottom (based on the theme.js headings in point 1). But we should be able to customise the component using props within components. There are so many variables: is this heading being used in a Content block, an InfoBanner, something else? I think we get the most user-friendly dev experience by importing the Heading component and customising to our needs. 
e.g. `<Heading level={2} marginTop="2rem">Example H2 heading with 32px of margin top</Heading>` – I think this is better than having to `const H2 = styled.h2....` etc in every page. 

What do you think?

**Review by ryancreatescopy — APPROVED** [review]

**Review by ryancreatescopy — APPROVED** [review]

**Comment by wackerow** [comment]
@ryancreatescopy I like that layout and explanation. I'd be in favor of merging this PR, but keeping the original issue open and tackling this with a new PR.

**Pull request merged**